beacon-chain/db/kv/blocks.go

```diff
@@ -134,6 +134,55 @@ func (s *Store) HasBlock(ctx context.Context, blockRoot [32]byte) bool {
 	return exists
 }
 
+// BlocksBySlot retrieves a list of beacon blocks and its respective roots by slot.
+func (s *Store) BlocksBySlot(ctx context.Context, slot uint64) ([]*ethpb.SignedBeaconBlock, bool, error) {
```

Comment:
It might be better to have the return values to follow this pattern;
```
(bool, []*ethpb.SignedBeaconBlock, error)
```
Roughly 
```
exists, blocks,err := s.BlocksBySlot
if err != nil {
}
if !exists {
.
.
.
```
This would be more in-line with how we perform similar checks across the repo.